rskj-core/src/main/java/co/rsk/pcc/bto/DeriveExtendedPublicKey.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.pcc.bto;
+
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.crypto.ChildNumber;
+import co.rsk.bitcoinj.crypto.DeterministicKey;
+import co.rsk.bitcoinj.crypto.HDKeyDerivation;
+import co.rsk.bitcoinj.crypto.HDUtils;
+import co.rsk.pcc.ExecutionEnvironment;
+import co.rsk.pcc.NativeContractIllegalArgumentException;
+import co.rsk.pcc.NativeMethod;
+import org.ethereum.core.CallTransaction;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * This implements the "deriveExtendedPublicKey" method
+ * that belongs to the BTOUtils native contract.
+ *
+ * @author Ariel Mendelzon
+ */
+public class DeriveExtendedPublicKey extends NativeMethod {
+    private final CallTransaction.Function function = CallTransaction.Function.fromSignature(
+            "deriveExtendedPublicKey",
+            new String[]{"string", "string"},
+            new String[]{"string"}
+    );
+
+    private final BTOUtilsHelper helper;
+
+    public DeriveExtendedPublicKey(ExecutionEnvironment executionEnvironment, BTOUtilsHelper helper) {
+        super(executionEnvironment);
+        this.helper = helper;
+    }
+
+    @Override
+    public CallTransaction.Function getFunction() {
+        return function;
+    }
+
+    @Override
+    public Object execute(Object[] arguments) {
+        String xpub = (String) arguments[0];
+        String path = (String) arguments[1];
+
+        NetworkParameters params = helper.validateAndExtractNetworkFromExtendedPublicKey(xpub);
+        DeterministicKey key;
+        try {
+            key = DeterministicKey.deserializeB58(xpub, params);
+        } catch (IllegalArgumentException e) {
+            throw new NativeContractIllegalArgumentException(String.format("Invalid extended public key '%s", xpub), e);
+        }
+
+        // Path must be of the form S, with S ::= n || n/S with n an unsigned integer
+        Pattern digitOnly = Pattern.compile("^\\d+$");
+        String[] pathChunks = path.split("/");
```

Comment:
this split will return a single value if the path is `0/` which is an invalid path, but when it goes through the digitOnly pattern will pass as a valid one. Curiously, if you send `/0` will split into `""`and `"0"` which will be properly rejected. 🤷‍♂️

Reply:
Good catch! Covered the case and fixed.